### DISCUSSION TITLE
Editor Toolkit State

### DISCUSSION THREAD

**Discussion by Matthew-w56** [rism-digital/verovio#3613] [Q&A]
Hello!

I really like your library, and I am in the process of using for a sheet music editing application.  But refeeding an entire score and re-rendering each edit is obviously much too slow (~1 second per page).  I notice that there is an edit functionality, to edit the internal representation directly.  That's great!  I have a few questions about it:

1) The documentation page says that it is still experimental.  Does it have enough functionality to support an editing application?  Is the experimental tag an out-of-date relic?

2) I can get it to delete a note (by reverse engineering the syntax expected), but when I try to render SVG from an edited score, it crashes.  At the same time, if I save the edited version to MEI and then load and render THAT to svg, it works just fine.  What might I be missing here?  Is it something to do with whatever "commit" does?

Thanks for the help!
Matthew Williams

My Code (written in Dart, through a FFI binding) (for reference):

  static final _binding = VerovioWrapper(DynamicLibrary.open("path-to-dll"));
  final _toolkitPointer = _binding.vrvToolkit_constructorResourcePath("path-to-resource-folder".toNativeUtf8().cast());
  String content = IOWorker.getFileText("normal-mei-file.mei");
  _binding.vrvToolkit_loadData(_toolkitPointer, _stringToCharPointer(content));
  verovio.executeEdit("{'action': 'delete', 'param': {'elementId': 'd1e966'}}");
  verovio.executeEdit("{'action': 'commit', 'param': {}}");
  // This svg render is what crashes the program
  String svgContent = _charPointerToString(_binding.vrvToolkit_renderToSVG(_toolkitPointer, 1, false));
  File file = File("my-desktop-path/alteredOutput.svg");
  file.writeAsString(svgContent);

**Comment by lpugin** [comment]
It seems that the editor functions are currently broken. This is experimental code, but I will still see if it can be fixed.

**Reply by Matthew-w56** [reply]
Okay.  And if it would be possible or help, I would be willing to help make a fix, if I can get briefed on what is going wrong.  Do you have an idea on what it might be?

**Comment by Matthew-w56** [comment]
Editor isn't super fleshed out and has a lot of bugs.  I fixed one of them with PR https://github.com/rism-digital/verovio/pull/3617
